airsonic-main/src/main/java/org/airsonic/player/security/CsrfSecurityRequestMatcher.java

```diff
@@ -20,24 +19,20 @@
  */
 @Component
 public class CsrfSecurityRequestMatcher implements RequestMatcher {
-    private Pattern allowedMethods = Pattern.compile("^(GET|HEAD|TRACE|OPTIONS)$");
-    private Collection<RegexRequestMatcher> whiteListedMatchers;
+    static private List<String> allowedMethods = Arrays.asList("GET", "HEAD", "TRACE", "OPTIONS");
+    private List<RegexRequestMatcher> whiteListedMatchers;
 
     public CsrfSecurityRequestMatcher() {
-        Collection<RegexRequestMatcher> whiteListedMatchers = new ArrayList<>();
-        whiteListedMatchers.add(new RegexRequestMatcher("/dwr/.*\\.dwr", "POST"));
-        whiteListedMatchers.add(new RegexRequestMatcher("/rest/.*\\.view(\\?.*)?", "POST"));
-        whiteListedMatchers.add(new RegexRequestMatcher("/search(?:\\.view)?", "POST"));
-        this.whiteListedMatchers = whiteListedMatchers;
+        this.whiteListedMatchers = Arrays.asList(
+            new RegexRequestMatcher("/dwr/.*\\.dwr", "POST"),
+            new RegexRequestMatcher("/rest/.*\\.view(\\?.*)?", "POST"),
+            new RegexRequestMatcher("/search(?:\\.view)?", "POST")
+        );
     }
 
     @Override
     public boolean matches(HttpServletRequest request) {
-
-        boolean skipCSRF =
-                allowedMethods.matcher(request.getMethod()).matches() ||
-                whiteListedMatchers.stream().anyMatch(matcher -> matcher.matches(request));
-
-        return !skipCSRF;
+        return !allowedMethods.contains(request.getMethod()) &&
+                whiteListedMatchers.stream().noneMatch(matcher -> matcher.matches(request));
```

Comment:
Tbh: I liked the previous boolean logic. It was easier to read than the double negative proposed here.

Reply:
I agree that the original logic was easier to read, but I also like getting rid of the temporary variable.  Maybe a decent compromise would be to use the original logic and just negate the whole thing on the return?

Reply:
Why is the temporary variable an issue?

Reply:
In fact I think the temporary variable is what improves the readability greatly.

Reply:
I just prefer to avoid using temp variables where possible.  But you are right, it does help readability, so is probably justified in this case.